great_expectations/expectations/core/expect_column_kl_divergence_to_be_less_than.py

```diff
@@ -288,11 +292,20 @@ def get_validation_dependencies(
                 #
                 # NOTE 20201117 - JPC - Would prefer not to include partition_metric_configuration here,
                 # since we have already evaluated it, and its result is in the kwargs for the histogram.
-                # However, currently the dependencies' configurations are not passed to the _validate method
+                # However, currently the validation_dependencies' configurations are not passed to the _validate method
                 #
-                dependencies["column.partition"] = partition_metric_configuration
-                dependencies["column.histogram"] = hist_metric_configuration
-                dependencies["column_values.nonnull.count"] = nonnull_configuration
+                validation_dependencies.set_metric_configuration(
+                    metric_name="column.partition",
```

Comment:
An effort for another time but could we make an enum for `metric_names`? I know this is an extensible part of our codebase but adding structure to our internally defined ones could be a nice win!

Reply:
@cdkini I am advocating a different approach -- it could be an `enum` on top of it; but I would like these metric names to not be strings, but `Python` references.  This would be an awesome hackathon project (which I have been proposing for quite some time now!).  There is also a Confluence document to that effect, if curious.  Thanks!